sound/soc/sof/ipc4-topology.c

```diff
@@ -1425,6 +1425,13 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	/* update pipeline memory usage */
 	sof_ipc4_update_pipeline_mem_usage(sdev, swidget, &process->base_config);
 	switch (process->process_type) {
+	case SOF_PROCESS_MICSEL:
+		memcpy(&micsel->output_format, micsel->available_fmt.out_audio_fmt,
```

Comment:
this makes no sense. Just outside the switch, you mentioned output format is not required and did not initialize it

Reply:
Different modules' requirements are different. For KPB module, we don't need output format; but for MICSEL module, we need output format. So we will not set output_format when calling sof_ipc4_init_audio_fmt() but will cope with the output format individually.